software/cryptotvgen/cryptotvgen/prepare_libs.py

```diff
@@ -85,8 +83,8 @@ def build_variants(variants, candidates_dir):
             try:
                 cp.check_returncode()
             except:
-                print(f'`{" ".join(cmd)}` failed! (exit code: {cp.returncode})')
-                sys.exit(1)
+                valid_str += f'`{" ".join(cmd)}` failed! (exit code: {cp.returncode})\n'
+        print(valid_str)
```

Comment:
I think the scope of this functionality is to be able to build all the libraries the user requested to be built (or all by default) among the supported available schemes (all NIST round 2 LWC candidates). I believe it should fail hard and stop the execution if and error was hit and not just print a message and continue.

Reply:
Am I using it wrong:
cryptotvgen --prepare_libs --candidates_dir=/home/rich/supercop-20200702/

I did this trying to build all of the libs. With this on my system it fails to build on  none round 2 candidates.

I cannot control what this tries to build. It just goes through and builds everything in both crypto_aead and crypto_hash.

Reply:
I would never be able to get all of the libs if this change in.

Reply:
> Am I using it wrong:
> cryptotvgen --prepare_libs --candidates_dir=/home/rich/supercop-20200702/
> 
> I did this trying to build all of the libs. With this on my system it fails to build on none round 2 candidates.
> 
> I cannot control what this tries to build. It just goes through and builds everything in both crypto_aead and crypto_hash.
You _can_ control what gets built through arguments to `--prepare_libs`. 
`cryptotvgen --prepare_libs pyjamask photonbeetle xoodyak --candidates_dir=/home/rich/supercop-20200702/` build _only_ all variants of `pyjamask`, `photonbeetle`, and `xoodyak` available in `/home/rich/supercop-20200702/`.
It's documented in [software/cryptotvgen/README.md](https://github.com/GMUCERG/LWC/blob/dev/software/cryptotvgen/README.md) as well as the built-in help.
Please let me know if the documentation is not fully clear or needs improvements. Also feel free to add your improvements to the docs.

If you just want to prepare libs on an unmodified version of SUPERCOP, the preferred way to run is just 
```
cryptotvgen --prepare_libs  # latest available SUPERCOP version
cryptotvgen --prepare_libs --supercop_version=20200702   # specific SUPERCOP version
cryptotvgen --prepare_libs pyjamask photonbeetle xoodyak # only variants of pyjamask, photonbeetle, xoodyak
```

Reply:
> I would never be able to get most of the libs if this change does not go in.

I really don't understand how this change is going to be helpful. If there are issues with building any of the LWC candidates please open a new issue.

Reply:
It used to try to build everything. That seems like a useful feature to keep.